File: app/src/main/java/com/azizbekian/spyur/utils/LogUtils.java
package com.azizbekian.spyur.utils;

import android.support.design.BuildConfig;
import android.util.Log;

/**
 * Created on April 02, 2016.
 *
 * @author Andranik Azizbekian (andranik.azizbekyan@gmail.com)
 */
public class LogUtils {

    private static final String PREFIX = "vvv";

    private static final boolean IS_DEBUGGABLE = BuildConfig.DEBUG;

    private LogUtils() {
        throw new RuntimeException("Unable to instantiate class " + getClass().getCanonicalName());
    }

    public static void wtf(String msg) {
        if (IS_DEBUGGABLE) Log.wtf(PREFIX, msg);
    }

    public static void i(String msg) {
        if (IS_DEBUGGABLE) Log.i(PREFIX, msg);
    }

    public static void e(String msg) {
        if (IS_DEBUGGABLE) Log.wtf(PREFIX, msg);
    }

    public static void wtf(int msg) {
        wtf("" + msg);
    }

    public static void i(int msg) {
        i("" + msg);
    }

    public static void e(int msg) {
        e("" + msg);
    }

    public static void wtf(boolean msg) {
        wtf("" + msg);
    }

    public static void i(boolean msg) {
        i("" + msg);
    }

    public static void e(boolean msg) {
        e("" + msg);
    }

    public static void wtf(float msg) {
        wtf("" + msg);
    }

    public static void i(float msg) {
        i("" + msg);
    }

    public static void e(float msg) {
        e("" + msg);
    }
}

File: app/src/main/kotlin/com/incentive/yellowpages/utils/LogUtils.kt
package com.incentive.yellowpages.utils

import com.incentive.yellowpages.BuildConfig
import timber.log.Timber

@Suppress("NOTHING_TO_INLINE")
class LogUtils private constructor() {

    init {
        throw RuntimeException("Unable to instantiate class " + javaClass.canonicalName)
    }

    companion object {

        val PREFIX = "vvv: "

        val IS_DEBUGGABLE = BuildConfig.DEBUG

        inline fun wtf(msg: String?) {
            if (IS_DEBUGGABLE) Timber.wtf(PREFIX + msg)
        }

        inline fun i(msg: String?) {
            if (IS_DEBUGGABLE) Timber.i(PREFIX + msg)
        }

        inline fun e(msg: String?) {
            if (IS_DEBUGGABLE) Timber.e(PREFIX + msg)
        }

        inline fun wtf(msg: Int?) {
            wtf("" + msg)
        }

        inline fun i(msg: Int?) {
            i("" + msg)
        }

        inline fun e(msg: Int?) {
            e("" + msg)
        }

        inline fun wtf(msg: Boolean?) {
            wtf("" + msg)
        }

        inline fun i(msg: Boolean?) {
            i("" + msg)
        }

        inline fun e(msg: Boolean?) {
            e("" + msg)
        }

        inline fun wtf(msg: Float?) {
            wtf("" + msg)
        }

        inline fun i(msg: Float?) {
            i("" + msg)
        }

        inline fun e(msg: Float?) {
            e("" + msg)
        }
    }
}
